src/bin/basic_subclass.rs

```diff
@@ -0,0 +1,185 @@
+//! # Basic Subclass example
+
+#[macro_use]
+extern crate glib;
+extern crate gio;
+extern crate gtk;
+
+extern crate once_cell;
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use glib::subclass;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use gtk::subclass::prelude::*;
+
+use gio::ApplicationFlags;
+
+use once_cell::unsync::OnceCell;
+
+#[derive(Debug)]
+struct WindowWidgets {
+    headerbar: gtk::HeaderBar,
+    label: gtk::Label,
```

Comment:
You're not using these anywhere other than initializing them. Might be good to do something with these widgets also, otherwise you don't need to store them :)

Reply:
Yea, need to find something to do with them I guess. Maybe add a button in the headerbar and turn the label in a counter.

Reply:
Sounds good